app/components/FilterPanel.tsx

```diff
@@ -0,0 +1,204 @@
+    filters: Filters;
+    setFilters: (f: Filters) => void;
+};
+
+export default function FilterPanel({ allOffers, filters, setFilters }: Props) {
+    // Extract unique brands and locations for filter dropdowns
+    const brands = Array.from(new Set(allOffers.map((o) => o.brand))).sort();
+    const locations = Array.from(
+        new Set(allOffers.map((o) => o.location))
+    ).sort();
+
+    return (
+        <aside className='mb-6 rounded border bg-white p-4 shadow-sm'>
+            <h3 className='mb-2 font-semibold'>Filtry (Elektronika)</h3>
+
+            <div className='grid gap-3 sm:grid-cols-2'>
+                <div>
+                    <label className='block text-xs text-gray-600'>
+                        Szukaj (tytuł)
+                    </label>
+                    <input
+                        value={filters.keyword ?? ''}
+                        onChange={(e) =>
+                            setFilters({
+                                ...filters,
+                                keyword: e.target.value || undefined,
+                            })
+                        }
+                        className='mt-1 w-full rounded border p-2 text-sm'
+                        placeholder='np. iPhone, Samsung...'
+                    />
+                </div>
+
+                <div>
+                    <label className='block text-xs text-gray-600'>Stan</label>
+                    <select
+                        value={filters.state ?? 'all'}
+                        onChange={(e) =>
+                            setFilters({
+                                ...filters,
+                                state:
+                                    (e.target.value as Filters['state']) ||
+                                    'all',
+                            })
+                        }
+                        className='mt-1 w-full rounded border p-2 text-sm'
+                    >
+                        <option value='all'>Wszystkie</option>
+                        <option value='new'>Nowe</option>
+                        <option value='used'>Używane</option>
+                    </select>
+                </div>
+
+                <div>
+                    <label className='block text-xs text-gray-600'>Marka</label>
+                    <select
+                        value={filters.brand ?? 'all'}
+                        onChange={(e) =>
+                            setFilters({
+                                ...filters,
+                                brand:
+                                    e.target.value === 'all'
+                                        ? undefined
+                                        : e.target.value,
+                            })
+                        }
+                        className='mt-1 w-full rounded border p-2 text-sm'
+                    >
+                        <option value='all'>Wszystkie</option>
+                        {brands.map((b) => (
```

Comment:
please use full words, this is production code not academic exercise 
```suggestion
                        {brands.map((brand) => (
```